src/server/game/World/World.cpp

```diff
@@ -2210,6 +2210,10 @@ void World::SetInitialWorldSettings()
     TC_LOG_INFO("server.loading", "Loading scenario poi data");
     sScenarioMgr->LoadScenarioPOI();
 
+    // We do it last to make sure everything a script might call in ShouldLoad() is initialized.
```

Comment:
This doesn't work, you need to process spell data before anything else starts reading SpellInfo, for example checking duration for loading creature_addon auras

Reply:
That makes sense, I honestly didn't think about it too much and just assumed nothing needed to access SpellInfo on load